### ISSUE TITLE
ConnectException: Failed to connect to firefoxusercontent.com/13.224.164.41:443

### ISSUE THREAD

**Issue by grigoryk** [mozilla-mobile/fenix#9530]
https://sentry.prod.mozaws.net/operations/fenix/issues/7591794/

```
ConnectException: Failed to connect to firefoxusercontent.com/13.224.164.41:443
    at org.mozilla.fenix.settings.SettingsFragment$updateAccountUIState$$inlined$let$lambda$1.invokeSuspend(SettingsFragment.kt:7)
...
(12 additional frame(s) were not displayed)

Failed to connect to firefoxusercontent.com/13.224.164.41:443
```

**Comment by grigoryk** [comment]
STR: app crashes when signed-in, offline, and opening settings.

**Comment by ebalazs-sv** [comment]
This is reproducible on Beta migration build 75.0.0-beta.2 (Build #2015730659) with Pixel 2 (Android 9) with the steps from https://github.com/mozilla-mobile/fenix/issues/9530#issuecomment-606394782.
Not reproducible on Nightly migration 200401 (Build #2015731955) with Pixel 2 (Android 9).

**Comment by ebalazs-sv** [comment]
Verified as fixed on Fennec Migration build 75.0.0-beta.3 4/1 with Pixel 2 (Android 9) and Huawei P9 Lite (Android 7), following the steps from https://github.com/mozilla-mobile/fenix/issues/9530#issuecomment-606394782.
